src/docker.rs

```diff
@@ -0,0 +1,167 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use docker_api::api::{ContainerCreateOpts, PullOpts, RegistryAuth, RmContainerOpts};
+use futures_util::{StreamExt, TryStreamExt};
+use std::path::PathBuf;
+use tar::Archive;
+
+use super::image;
+
+pub struct DockerImage {
+    pub image: String,
+    pub repo: String,
+    pub tag: String,
+    pub runtime: docker_api::Docker,
+}
+
+#[async_trait]
+impl image::Image for DockerImage {
```

Comment:
nit...I wonder if we can rename `image::Image` as the trait, since it's redundant. Maybe something like `core::Image` makes more sense?

Reply:
I do like that better, I'll work to get that working.